packages/core/src/services/documents/create.ts

```diff
@@ -1,16 +1,12 @@
-import {
-  DocumentVersion,
-  documentVersions,
-  getDocumentsAtCommit,
-  Result,
-  Transaction,
-} from '@latitude-data/core'
+import { DocumentVersion, Result } from '@latitude-data/core'
 import { BadRequestError } from '$core/lib/errors'
 
 import {
-  assertCommitIsEditable,
-  existsAnotherDocumentWithSamePath,
-} from './utils'
+  getDraft,
+  getMergedAndDraftDocuments,
+  replaceCommitChanges,
+  resolveDocumentChanges,
+} from './shared'
 
 export async function createNewDocument({
   commitId,
```

Comment:
in order to ensure tenancy pass a commit instance instead of instance id, this will force the consumer to ensure tenancy before calling any writer (and facilitate the refactor [here](https://github.com/latitude-dev/latitude-llm/pull/51))